### PR TITLE
Documentation compliance issues

### LINKED ISSUE #1331: Documentation compliance issues
https://github.com/BHoM/Revit_Toolkit/pull/1320/checks?check_run_id=12327193012

Please fix documentation compliance issues noted in this check run.

### PR DESCRIPTION
**Pull request by travispotterBH**
### Issues addressed by this PR

Closes #1331 

### Changelog

- Compliance issues addressed.

### CODE REVIEW

**Comment by travispotterBH** [comment]
Hey @BHoMBot check documentation-compliance for me please. I've made a few changes, and want to easily see what is left to resolve. Thanks 😄

**Comment by travispotterBH** [comment]
Hey @BHoMBot check documentation-compliance for me please. I've made a few changes, and want to easily see what is left to resolve. Thanks 😄

**Comment by travispotterBH** [comment]
@pawelbaran please check for correctness of changes to inputs, descriptions, and in particular please review the addition of `throw NotImplementedException` on these methods below:
https://github.com/BHoM/Revit_Toolkit/blob/26d45c5ffba16e962551813306ca0f12a1c9033e/Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs#L115-L118

**Comment by travispotterBH** [comment]
@BHoMBot  would you be so kind as to check required?

**Comment by travispotterBH** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance

File: Revit_Core_Adapter/Listener/Forms/UpdatePortsForm.cs [hunk 1]
```diff
@@ -33,6 +35,8 @@ public partial class UpdatePortsForm : Form
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Window for updating ports used in the Revit Listener.")]
```
Comment (pawelbaran) [inline Revit_Core_Adapter/Listener/Forms/UpdatePortsForm.cs:38]:
```suggestion
        [Description("Constructor of window for updating ports used by the Revit Listener.")]
```

File: Revit_Core_Adapter/RevitListenerAdapter.cs [hunk 2]
```diff
@@ -44,7 +46,10 @@ public partial class RevitListenerAdapter : BHoMAdapter,  IInternalRevitAdapter
         /***************************************************/
         /****            Public Constructors            ****/
         /***************************************************/
-        
+        [Description("Revit listener adapter constructor.")]
```
Comment (pawelbaran) [inline Revit_Core_Adapter/RevitListenerAdapter.cs:49]:
```suggestion
        
        [Description("Revit listener adapter constructor.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 3]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
         /****              Public methods               ****/
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:54]:
```suggestion
        [Description("Method returning RenderNodeAction to be taken at the start of parsing an ElementNode.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 4]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "ElementNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnElementBegin2D(ElementNode node)
         {
             return base.OnElementBegin(node.ElementId);
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:64]:
```suggestion
        [Description("Method to be executed at the end of parsing an ElementNode.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 5]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "ElementNode to be processed.")]
         public void OnElementEnd2D(ElementNode node)
         {
+            throw new NotImplementedException();
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:68]: This method is empty on purpose: it says 'do nothing if you get here', which is very different from 'throw an exception if you get here' 😉 so please bring it back to the original state - refers to all cases where you have added the exception.

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 6]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:73]:
```suggestion
        [Description("Method returning RenderNodeAction to be taken against an edge of a face.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 7]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "FaceEdgeNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnFaceEdge2D(FaceEdgeNode node)
         {
             return RenderNodeAction.Skip;
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:83]:
```suggestion
        [Description("Method returning RenderNodeAction to be taken against a face silhouette.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 8]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "FaceSilhouetteNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnFaceSilhouette2D(FaceSilhouetteNode node)
         {
             return RenderNodeAction.Skip;
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:93]:
```suggestion
        [Description("Method returning RenderNodeAction to be taken against a curve.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 9]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "CurveNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnCurve(CurveNode node)
         {
             return RenderNodeAction.Skip;
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:103]:
```suggestion
        [Description("Method returning RenderNodeAction to be taken against a polyline.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 10]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Input("node", "PolylineNode to be processed.")]
+        [Output("action", "RenderNodeAction.")]
         public RenderNodeAction OnPolyline(PolylineNode node)
         {
             return RenderNodeAction.Skip;
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:113]:
```suggestion
        [Description("Method to be executed against a line segment.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 11]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Description("Method used to return a RenderNodeAction.")] 
+        [Input("segment", "LineSegment to be processed.")]
         public void OnLineSegment(LineSegment segment)
         {
+            throw new NotImplementedException();
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:122]:
```suggestion
        [Description("Method to be executed against a polyline segment.")]
```

File: Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs [hunk 12]
```diff
@@ -46,61 +51,88 @@ public Active2dViewVisibilityContext(Document hostDocument, Document targetDocum
...
+        [Description("Method used to return a RenderNodeAction.")] 
+        [Input("segments", "PolylineSegments to be processed.")]
         public void OnPolylineSegments(PolylineSegments segments)
         {
+            throw new NotImplementedException();
         }
 
         /***************************************************/
 
+        [Description("Method used to return a RenderNodeAction.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/Active2dViewVisibilityContext.cs:131]:
```suggestion
        [Description("Method to be executed against a text.")]
```

File: Revit_Core_Engine/Objects/WarningSwallower.cs [hunk 13]
```diff
@@ -30,7 +32,9 @@ public class WarningSwallower : IFailuresPreprocessor
         /***************************************************/
         /****              Public methods               ****/
         /***************************************************/
-
+        [Description("Failure message swallower, used to continue the process without regard to errors.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Objects/WarningSwallower.cs:35]: Your text should be the description of a class itself.

Here I would put something along the lines:
```suggestion
        
        [Description("Preprocess the failures, in this case swallow all of them.")]
```

File: Revit_Core_Engine/Query/ElementType.cs [hunk 14]
```diff
@@ -147,10 +147,13 @@ public static RebarBarType ElementType(this oM.Physical.Reinforcement.IReinforci
...
         [Input("document", "Revit document to parse in search for the element type.")]
+        [Input("builtInCategories", "List of built in categories to be considered when searching for the element type.")]
         [Input("settings", "Revit adapter settings to be used while performing the query.")]
         [Output("attributes", "Revit host object attributes to be used when converting the input BHoM object to Revit.")]
-        public static HostObjAttributes ElementType(this oM.Physical.Constructions.IConstruction construction, Document document, IEnumerable<BuiltInCategory> builtInCategories, RevitSettings settings = null)
-        {
+        public static HostObjAttributes ElementType(this
+        oM.Physical.Constructions.IConstruction construction, Document
+        document, IEnumerable<BuiltInCategory> builtInCategories, RevitSettings
+        settings = null) {
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/ElementType.cs:156]: I am not a big fan of splitting method parameters into separate lines, and I am even less of a fan of putting the entry bracket at the end of preceding line (maybe this is why I have never learnt js 🙈)

File: Revit_Core_Engine/Query/RenderMeshes.cs [hunk 15]
```diff
@@ -41,6 +41,7 @@ public static partial class Query
         [Input("element", "Element to display thorugh RenderMeshes.")]
         [Input("options", "Revit API options for the retrieval, including e.g. `IncludeNonVisibleObjects`.")]
         [Input("settings", "Revit Adapter settings to be applied in retrieving the meshes corresponding to the element.")]
+        [Output("renderMeshes", "RenderMesh objects corresponding to the input Revit Element.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/RenderMeshes.cs:44]:
```suggestion
        [Output("renderMeshes", "RenderMesh objects representing the input Revit Element.")]
```

File: Revit_Core_Engine/Query/RenderMeshes.cs [hunk 16]
```diff
@@ -41,6 +41,7 @@ public static partial class Query
         [Input("element", "Element to display thorugh RenderMeshes.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/RenderMeshes.cs:41]:
```suggestion
        [Input("element", "Element to display through RenderMeshes.")]
```

**Review by pawelbaran — CHANGES_REQUESTED** [review]

**Comment by travispotterBH** [comment]
@BHoMBot  check required

**Comment by travispotterBH** [comment]
@bhombot check copyright-compliance
@BHoMBot  check dataset-compliance

**Review by pawelbaran — APPROVED** [review]
Happy with the changes now 👍

**Pull request merged**